scripts/bot-deployment/DeployBotGCP.sh

```diff
@@ -10,24 +10,10 @@ fi
 
 echo "🔥 Starting deployment script for bot" $1
 
-# Bot names are <identifer>-<network>-<bot-type>. EG: ethbtc-liquidator-mainnet.
-# This cut removes the network as this is not included in a service account. ethbtc-liquidator-mainnet becomes ethbtc-liquidator
-paramToServiceAccount=$(echo $1 | cut -d'-' -f 1,3)
-serviceAccountEmail=""
+# This service account has all permissions needed for any bot to run.
+serviceAccountEmail="emp-bot@uma-protocol.iam.gserviceaccount.com"
```

Comment:
I'm opposed to hardcoding these values like this. The point of these scripts is to let any user who runs the script deploy to _their_ respective GCP instances. I would be ideal in my mind to still pull these from GCP such that it is somewhat dynamic.

At the same time though the scripts do assume a lot from the user: 1) correctly named and stored config file and 2) correctly named service account. For this reason, I guess it's ok. I'm just not super happy with exposing it like this.

@mrice32 what do you think?

Reply:
How would we pull it dynamically without hardcoding it?

Reply:
I think what @chrismaree is referring to is making some call like `gcloud iam service-accounts list` and using that to find the right service account somehow. Since we're basically only using one service account for our bots, I think we have a few options:
1. Don't hardcode _our_ project, but do hardcode the service account name. You could do this by getting the result of the command:
```
gcloud iam service-accounts list --filter name:emp-bot --format 'value(email)'
```

As an added bonus, if that didn't return anything, the script can give the user a useful message that that service account doesn't exist. This would be similar to the error message that was removed from this script.

2. Have it hardcoded, but make it easy to override by providing an argument. This would look something like:
```bash
BOT_NAME=$1
# Sets to the default value if unset.
: ${BOT_NAME:=emp-bot@uma-protocol.iam.gserviceaccount.com}
```

3. Require the user to pass in the service account name.
4. Maybe a combination of 1 and 2, where the user can override the service account name that's passed into the gcloud command.

I get where Chris is coming from -- I think we're okay with making these scripts somewhat particular to this setup, but we might not want them to be impossible to use outside of our google cloud org/team. However, at this point, I'm not entirely convinced someone else is going to use our GCP scripts, so I'm not super opinionated either way.

Reply:
Yeah my opinion is that there is a lot of setup to get GCP configured exactly like our own right now. 

Also @mrice32 's proposal still hard-codes the "emp-bot" name filter

Reply:
@nicholaspai yep. I think the difference is that you could create an `emp-bot` service account in any GCP project.

Totally get where you're coming from tho.

Reply:
However, I do think being able to check for the existence of the service account before deploying is useful.

Reply:
Unless GCP gives a similar error message if you use an invalid service account.